packages/react-instantsearch-hooks-web/src/ui/RangeInput.tsx

```diff
@@ -132,9 +132,13 @@ export function RangeInput({
             step={step}
             placeholder={min?.toString()}
             disabled={disabled}
-            onInput={({ currentTarget }) =>
-              setRange({ from: Number(currentTarget.value), to })
-            }
+            onInput={({ currentTarget }) => {
+              const value = currentTarget.value;
+              setRange({
+                from: value ? Number(value) : unsetNumberInputValue,
```

Comment:
how does this prevent invalid (1.) Number turning to NaN?

Reply:
I'm not sure, but I'd guess setting the state to an empty string makes the number input temporarily uncontrolled?

Reply:
I think it might make sense to use the "string storage" in both React InstantSearch Hooks and InstantSearch.js, as it's more consistent

Reply:
Done in https://github.com/algolia/instantsearch/pull/5541/commits/e661ab203a1dfb1b07f82c74b866a5666354cce3